source/client/RunManagement.py

```diff
@@ -0,0 +1,250 @@
+        temp_run_group = card_group
+        card_group = []
+        card_group.append(temp_run_group[0])
+        for indx in range(len(temp_run_group)-1):
+            card = temp_run_group[indx+1]
+            if temp_run_group[indx].tempnumber == card.tempnumber + 1:
+                card_group.append(card)
+            elif temp_run_group[indx].tempnumber == card.tempnumber + 2:
+                if len(groups_wilds) > 0:
+                    this_wild.pop(groups_wilds[0])
+                    card_group.append(this_wild)
+                    card_group.append(card)
+                else:
+                    raise Exception('too big a gap between numbers')
+
+    #  Review note - Handle Aces after other cards, else ran into problem when wanted Ace, Joker, 3...
+    # Rare for Ace Hi and Ace low to both be options. If they are, does it make a difference which one they are?
+    # If run is A?,2,Wild,4...J,Q,K,A? then it does (wild could still be played above King, but not below 2).
+    if len(aces_list) > 0:
+        if len(aces_list) > 2:
+            raise Exception('Cannot play more than 2 Aces in a single run.')
+        # Possible Ace(s) replacing a wild card with tempnumber 13 or -1.
+        # If have wild in Ace slot then remove it from run.  It will be placed back in run after Aces placed.
+        if isWild(card_group[0], wild_numbers) and abs(card_group[0].tempnumber) == 1:
+            this_wild = card_group.pop(0)
+            groups_wilds.append(this_wild)
+        if isWild(card_group[-1], wild_numbers) and card_group[-1].tempnumber == 13:
+            this_wild = card_group.pop(-1)
+            groups_wilds.append(this_wild)
+        # Can Ace go high but not low?
+        if card_group[-1].tempnumber == 13 and not card_group[0].tempnumber == 2:
+            this_ace = aces_list.pop(0)
+            this_ace.tempnumber = 14
+            card_group.append(this_ace)
+            # Can Ace go low but not high?
+        elif not card_group[-1].tempnumber == 13 and card_group[0].tempnumber == 2:
+            this_ace = aces_list.pop(0)
+            this_ace.tempnumber = -1
+            card_group.insert(0, this_ace)
+        elif card_group[-1].tempnumber == 13 and card_group[0].tempnumber == 2:
+            if len(aces_list) == 1:
+                print("IF ACE CAN BE HIGH OR LOW THAN AUTOMATICALLY MAKING IT LOW, RATHER THAN HANDLING CORNER CASE.")
+                this_ace = aces_list.pop(0)
+                this_ace.tempnumber = -1
+                card_group.insert(0, this_ace)
+            elif len(aces_list) == 2:
+                this_ace = aces_list.pop(0)
+                this_ace.tempnumber = -1
+                card_group.insert(0, this_ace)
+                this_ace = aces_list.pop(0)
+                this_ace.tempnumber = 14
+                card_group.append(0, this_ace)
+    if len(aces_list) > 0 and len(groups_wilds) > 0:
+        # Can Aces that previously could not be played now be played?
```

Comment:
I think this would be more efficient if you made the 'place remaining wilds' helper separate then you could call it before placing aces, and place aces allowing for replacing wilds at the ends, and then rerun the 'place remaining wilds' helper - I think that would be easier to understand and have better code reuse along with probably catching the 'too many wilds' issue faster.